daktari/checks/direnv.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+from daktari.command_utils import get_stdout
+
+from daktari.os import OS
+from daktari.check import Check, CheckResult
+from daktari.version_utils import get_simple_cli_version
+from typing import Optional
+
+
+class DirenvInstalled(Check):
+    name = "direnv.installed"
+
+    def __init__(self, required_version: Optional[str] = None, recommended_version: Optional[str] = None):
+        self.required_version = required_version
+        self.recommended_version = recommended_version
+        self.suggestions = {
+            OS.OS_X: "<cmd>brew install direnv</cmd>",
+            OS.GENERIC: "Install direnv: https://direnv.net/#getting-started",
+        }
+
+    def check(self) -> CheckResult:
+        installed_version = get_simple_cli_version("direnv")
+        return self.validate_semver_expression(
+            "direnv", installed_version, self.required_version, self.recommended_version
+        )
+
+
+class DirenvAllowed(Check):
+    name = "direnv.allowed"
+    depends_on = [DirenvInstalled]
+
+    def __init__(self, parent_file_name: Optional[str] = None):
+        self.file_name = parent_file_name
```

Comment:
What's the thinking behind specifying `parent_file_name`?

Reply:
Ah I was just trying it to see if that fixed a `mypy` error in CI (it didn't)

It was actually because I set `self.file_name` but called `self.parent_file_name`

Reply:
Okay, sorted it out in [23c92da](https://github.com/glean-notes/daktari/pull/69/commits/23c92daaaf26de0871e102623a8e059d293e205d), hopefully it all makes sense now 🤞